test/Microsoft.Azure.Devices.Edge.Test.Common/YamlDocument.cs

```diff
@@ -2,34 +2,35 @@
 
 namespace Microsoft.Azure.Devices.Edge.Test.Common
 {
-    using System.Collections.Generic;
     using System.IO;
     using System.Linq;
     using YamlDotNet.Serialization;
 
+    using Node = System.Collections.Generic.Dictionary<object, object>;
+
     class YamlDocument
     {
-        readonly Dictionary<object, object> root;
+        readonly Node root;
 
         public YamlDocument(string input)
         {
             var reader = new StringReader(input);
             var deserializer = new Deserializer();
-            this.root = (Dictionary<object, object>)deserializer.Deserialize(reader);
+            this.root = (Node)deserializer.Deserialize(reader);
         }
 
         public void ReplaceOrAdd(string dottedKey, string value)
         {
-            Dictionary<object, object> node = this.root;
+            Node node = this.root;
             string[] segments = dottedKey.Split('.');
             foreach (string key in segments.SkipLast(1))
             {
                 if (!node.ContainsKey(key))
                 {
-                    node.Add(key, new Dictionary<object, object>());
+                    node.Add(key, new Node());
                 }
 
-                node = (Dictionary<object, object>)node[key];
+                node = (Node)node[key];
```

Comment:
it is possible that node[key] is not type of Node, therefore variable "node" will be null; and it will throw exception in line 30 when calling ContainsKey.  right?

Reply:
Our config.yaml file is composed of key: value mappings; we don't have any lists or scalars. Given that, each node in the document will always deserialize into a dictionary.

If the config.yaml fail is malformed this line could throw, but that's the behavior we want.

If our schema was updated to support lists or scalars, then this library would need to be updated as well. I can add a comment to this effect.

Reply:
Done.